### DISCUSSION TITLE
如何实现随着节点一起放大、缩小和旋转的工具 node tool

### DISCUSSION THREAD

**Discussion by MrMengJ** [antvis/X6#2045] [Q&A]
![image](https://user-images.githubusercontent.com/26325346/164953494-a9f3801c-a72f-4539-bcf7-8b6092432b5b.png)
想实现一个自定义节点工具，类似于内置的button工具一样。但是需要满足两个条件：

1. 可以随着节点的放大缩小一起同比例放大缩小，随着节点的旋转一起跟着旋转，跟着画布的resizing缩放一起缩放，等等。就是要保持跟节点的相对位置，同步变化。
2. 自定义工具需要使用 html 元素来实现

请问一下，有没有具体的实现思路指教一下

**Comment by z7x0** [comment]
怎么实现呀？
